src/kbc_modules/offline_fs_kbc/common.rs

```diff
@@ -50,9 +51,23 @@ pub fn load_keys(keyfile_name: &str) -> Result<Keys> {
         .collect()
 }
 
+pub fn load_resources(resources_file_name: &str) -> Result<Resources> {
+    let resources_json = fs::read_to_string(resources_file_name)?;
+    // Redact parsing errors to avoid side-channels
+    let encoded_resources: HashMap<String, String> = serde_json::from_str(&resources_json)
+        .map_err(|_| anyhow!("Failed to parse resources JSON file"))?;
+    encoded_resources
+        .iter()
+        .map(|(k, v)| match decode(v) {
+            Ok(resource) => Ok((k.clone(), resource)),
+            Err(_) => Err(anyhow!("Failed to decode resource")),
+        })
+        .collect()
+}
+
 pub mod tests {
     pub use super::*;
-
+    use crate::kbc_modules::ResourceName::*;
```

Comment:
I would change this to `use crate::kbc_modules::ResourceName;` and then use `ResourceName::foo` below. I think that would be a bit clearer.

Reply:
updated to
```
    use crate::kbc_modules::ResourceName::Policy;
    use crate::kbc_modules::ResourceName::SigstoreConfig;
    use crate::kbc_modules::ResourceName::GPGPublicKey;
```